main.cpp

```diff
@@ -52,12 +57,21 @@ NanostackRfPhyMcr20a rf_phy(MCR20A_SPI_MOSI, MCR20A_SPI_MISO, MCR20A_SPI_SCLK, M
 #endif //MBED_CONF_APP_RADIO_TYPE
 #endif //MESH
 
-#ifndef MESH
-// This is address to mbed Device Connector
-#define MBED_SERVER_ADDRESS "coap://api.connector.mbed.com:5684"
+#if defined MESH
+#define IPV6
+#elif MBED_CONF_LWIP_IPV6_ENABLED==true
+#define IPV6
 #else
-// This is address to mbed Device Connector
-#define MBED_SERVER_ADDRESS "coaps://[2607:f0d0:2601:52::20]:5684"
+#define IPV4
+#endif
+
+#ifdef IPV6
+    // This is IPv6 address to mbed Device Connector
+    // mesh is always IPv6
+    #define MBED_SERVER_ADDRESS "coaps://[2607:f0d0:2601:52::20]:5684"
+#else
+    // This is IPv4 address to mbed Device Connector
+    #define MBED_SERVER_ADDRESS "coap://api.connector.mbed.com:5684"
```

Comment:
I this was explained by @SeppoTakalo , this part is to avoid DNS request with MESH (which is not supported). This change should be reverted, but instead we could comment here the reason for this if/else